LSPS0/README.md

```diff
@@ -0,0 +1,603 @@
+[BOLT1]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/01-messaging.md
+
+LSPs MUST set the `features` bit numbered 729
+(`option_supports_lsps`) in both the `init` message on connection
+establishment, and in their own advertised `node_announcement`.
+Clients MUST NOT set `features` bit numbered 729 in either
+context.
+
+> **Rationale** As clients are the one who must initiate any
+> message ID 37913 messages before LSPs respond, they need to
+> separately discover whether a peer they connect to is
+> indeed an LSPS LSP that recognizes that message.
+> Broadcasting this as part of gossip allows a client to
+> discover new LSPs by simply downloading gossip.
+> The bit 729 was chosen by random and has no special meaning.
+
+### LSPS Specification Support Query
+
+The client can determine if an LSP supports a particular LSPS
+specification other than LSPS0 via the `method` named
+`lsps0.listprotocols`, which has the following parameters:
+
+```JSON
+{
+  "comment": "This comment is ignored"
+}
+```
+
+`comment` is an optional string, and is ignored by the LSP.
+It defaults to the empty string.
+
+> **Rationale** In addition to allowing clients to determine what
+> protocols an LSP supports, this query is also an example of what
+> other LSPS specifications need to describe, such as the parameters
+> of an API endpoint, the type of each parameter, whether each
+> parameter is optional, the default value if the parameter is
+> optional but not specified by the client, and what each parameter
+> means for the LSP.
+
+`lsps0.listprotocols` has no errors defined.
+
+The response datum is an object like the below:
+
+```JSON
+{
+  "protocols": [1, 3]
+}
+```
+
+`protocols` is an array of numbers, indicating the LSPS specification
+number for the LSPS specification the LSP supports.
+LSPs do not advertise LSPS0 support and 0 MUST NOT appear in the
+`protocols` array.
+
+> **Rationale** LSPS0 support is advertised via `features` bit 729
+> already, so specifying `0` here is redundant.
+
+> **Non-normative** The example below would not be necessary for other
+> LSPS specifications, but gives an idea of how the JSON-RPC 2.0
+> protocol would look like, when using this API endpoint.
+
+As a concrete example, a client might send the JSON object below
+inside a message ID 37913 in order to query what LSPS protocols
+the LSP supports:
+
+```JSON
+{
+  "method": "lsps0.listprotocols",
+  "jsonrpc": "2.0",
+  "id": "example#3cad6a54d302edba4c9ade2f7ffac098",
+  "params": {}
+}
+```
+
+The LSP could then respond with a message ID 37913 with the following
+payload, indicating it supports LSPS1 and LSPS3 (in addition to LSPS0):
+
+```JSON
+{
+  "jsonrpc": "2.0",
+  "id": "example#3cad6a54d302edba4c9ade2f7ffac098",
+  "result": {
+    "protocols": [1, 3],
+    "example-undefined-key-that-clients-should-ignore": true
+  }
+}
+```
+
+## Notes On Implementation
```

Comment:
I think it is helpful for the initial version of the spec to have it at least, so that implementors can quickly get up to speed.  We can remove it later if details change enough that maintaining it becomes onerous.